packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,191 +1,71 @@
-   */
-  readonly onClick: () => void;
-
-  /**
-   * Helper text displayed for press action
-   */
-  readonly label: string;
-}
-
-const ComboboxContext = React.createContext(
-  {} as { open: boolean; setOpen: (open: boolean) => void },
-);
 
-const ComboboxContextProvider = (props: ComboboxProps) => {
-  const [open, setOpen] = useState<boolean>(false);
-  const contentRef = useRef<HTMLDivElement>(null);
-
-  useEffect(() => {
-    function handleClickOutside(event: MouseEvent) {
-      if (
-        contentRef.current &&
-        !contentRef.current.contains(event.target as Node)
-      ) {
-        setOpen(false);
-      }
-    }
-
-    document.addEventListener("mousedown", handleClickOutside);
-    return () => {
-      document.removeEventListener("mousedown", handleClickOutside);
-    };
-  }, [contentRef]);
-
-  return (
-    <ComboboxContext.Provider value={{ open, setOpen }}>
-      <div className={styles.combobox} ref={contentRef}>
-        {props.children}
-      </div>
-    </ComboboxContext.Provider>
+export const Combobox = (props: ComboboxProps): JSX.Element => {
+  const { renderContent, renderTrigger } = useComboboxValidation(
+    props.children,
   );
-};
-
-export const Combobox = (props: ComboboxProps) => {
-  return <ComboboxContextProvider>{props.children}</ComboboxContextProvider>;
-};
-
-const TriggerButton = (props: TriggerProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
-
-  return <Button label={props.label} onClick={() => setOpen(!open)} />;
-};
-
-const TriggerChip = (props: TriggerProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
   return (
-    <div
-      className={styles.trigger}
-      onClick={() => {
-        setOpen(!open);
-      }}
-    >
-      <Text>{props.label}</Text>
-      <div className={styles.triggerIcon}>
-        <Icon name="add" size="small" />
-      </div>
-    </div>
+    <ComboboxContextProvider>
+      {renderTrigger}
+      {renderContent}
+    </ComboboxContextProvider>
   );
 };
 
-const Content = (props: ContentProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
-
-  const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
-  );
-
-  function handleSearch(event: React.ChangeEvent<HTMLInputElement>) {
-    setSearchValue(event.target.value);
-  }
+function useComboboxValidation(children: ReactNode): {
+  renderTrigger: ReactNode;
+  renderContent: ReactNode;
+} {
+  const childrenArray = React.Children.toArray(children);
+  let renderTrigger: ReactNode, renderContent: ReactNode;
+
+  childrenArray.forEach(child => {
+    if (isTriggerElement(child)) {
+      if (renderTrigger) {
+        useAssert(true, "Combobox can only have one Trigger element");
```

Comment:
in retrospect, `useAssert` should probably not be a hook (:past-derl:). But since this starts with `use` we'll treat it like a hook and say we's breaking dem [hook rulez](https://legacy.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level)

> Don’t call Hooks inside loops, conditions, or nested functions

Reply:
true true true. kinda surprised TS didn't yell at me about that one.

ohhh I see how it's being used with conditions now! I just happened to look at all the usages where it was just `true` as the first arg, but you can obviously have it evaluate an expression there and then either throw it or not throw it

I'll update that, and yeah I agree. you could make a good argument for this being a util rather than a hook for what it needs to do. I don't think it being a hook adds a whole lot other than limitation of how it can be used.

Reply:
yeah, we don't have the linter for that yet. we ain't as fancy as the mobile haha

Reply:
k I've stopped calling this conditionally and in a loop, although I'm not convinced the latter is entirely necessary

https://stackoverflow.com/questions/53906843/why-cant-react-hooks-be-called-inside-loops-or-nested-function#comment94658381_53906907
https://stackoverflow.com/questions/53546553/is-foreach-guaranted-to-loop-in-the-same-order-everytime-over-the-same-data-set

I'm pretty sure the order will always be respected here and we could get away with keeping it in the loop, but whatever moving it outside is trivial - the only downside is a bit uglier code imo